### PR TITLE
Use Update instead of Insert for creative_ads_conversion table

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/20191

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See https://github.com/brave/brave-browser/issues/20191

### CODE REVIEW

**Review by emerick — APPROVED** [review]
LGTM

File: vendor/bat-native-ads/src/bat/ads/internal/database/tables/conversions_database_table.cc [hunk 1]
```diff
@@ -259,6 +264,46 @@ void Conversions::MigrateToV10(mojom::DBTransaction* transaction) {
...
 }
 
+void Conversions::MigrateToV20(mojom::DBTransaction* transaction) {
+  DCHECK(transaction);
+
+  util::Rename(transaction, "creative_ad_conversions",
+               "creative_ad_conversions_temp");
+
+  const std::string& query = base::StringPrintf(
+      "CREATE TABLE ad_conversions "
```
Comment (aseren) [inline vendor/bat-native-ads/src/bat/ads/internal/database/tables/conversions_database_table.cc:274]: should it be `creative_ad_conversions` table ?

Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/database/tables/conversions_database_table.cc]: Great spot,thanks

Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/database/tables/conversions_database_table.cc]: Fixed

**Review by aseren — APPROVED** [review]

File: vendor/bat-native-ads/src/bat/ads/internal/database/tables/conversions_database_table.cc [hunk 2]
```diff
@@ -259,6 +265,41 @@ void Conversions::MigrateToV10(mojom::DBTransaction* transaction) {
   util::CreateIndex(transaction, "creative_ad_conversions", "creative_set_id");
 }
 
+void Conversions::MigrateToV20(mojom::DBTransaction* transaction) {
+  DCHECK(transaction);
+
+  const std::string& temp_table_name = "creative_ad_conversions_temp";
+
+  const std::string& query =
+      "CREATE TABLE creative_ad_conversions_temp "
```
Comment (moritzhaller) [inline vendor/bat-native-ads/src/bat/ads/internal/database/tables/conversions_database_table.cc:274]: nit: why not interpolate with `temp_table_name` from L271

**Review by moritzhaller — APPROVED** [review]
LGTM

File: vendor/bat-native-ads/src/bat/ads/internal/database/tables/conversions_database_table.cc (hunk 2 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/database/tables/conversions_database_table.cc:274]: Migration will be refactored as part of https://github.com/brave/brave-browser/issues/14806

**Comment by btlechowski** [comment]
Verification passed on

Brave | 1.37.6 Chromium: 98.0.4758.74 (Official Build) nightly (64-bit)
-- | --
Revision | d0fe1ec4df090cd3eb02b591228505e12ea476e9-refs/branch-heads/4758@{#935}
OS | Ubuntu 18.04 LTS


Reproduced the issue on 1.34.81
After `urlPattern` was changed and catalog updated, there were duplicate entries for the same creativeSetID:
image

In 1.37.x, the url pattern was updated
Before update:
image

After update:
image

**Pull request merged**